src/core/ignore.c

```diff
@@ -326,26 +322,23 @@ static void ignore_remove_config(IGNORE_REC *rec)
 
 static void ignore_init_rec(IGNORE_REC *rec)
 {
-#ifdef HAVE_REGEX_H
-	char *errbuf;
-	int errcode, errbuf_len;
+	if (rec->regexp_compiled) {
+		g_regex_unref(rec->preg);
+		rec->regexp_compiled = FALSE;
+	}
 
-	if (rec->regexp_compiled) regfree(&rec->preg);
-	rec->regexp_compiled = FALSE;
 	if (rec->regexp && rec->pattern != NULL) {
-		errcode = regcomp(&rec->preg, rec->pattern,
-				REG_EXTENDED|REG_ICASE|REG_NOSUB);
-		if (errcode != 0) {
-			errbuf_len = regerror(errcode, &rec->preg, 0, 0);
-			errbuf = g_malloc(errbuf_len);
-			regerror(errcode, &rec->preg, errbuf, errbuf_len);
-			g_warning("Failed to compile regexp '%s': %s", rec->pattern, errbuf);
-			g_free(errbuf);
+		GError *re_error;
+
+		rec->preg = g_regex_new(rec->pattern, G_REGEX_CASELESS | G_REGEX_EXTENDED, 0, &re_error);
```

Comment:
This doesn't mean the same thing as `REG_EXTENDED`

>If a pattern is compiled with the G_REGEX_EXTENDED option, whitespace in the pattern (other than in a character class) and characters between a # outside a character class and the next newline are ignored. An escaping backslash can be used to include a whitespace or # character as part of the pattern. 

For posix regex `REG_EXTENDED` means using the extended flavor instead of the basic one (so irssi defaults to not have to escape `(|)` and others)